### PR TITLE
[QNN] llama model SNR test

### PR DESCRIPTION
**Pull request by YIWENX14**
### Summary
[PLEASE REMOVE] See [CONTRIBUTING.md's Pull Requests](https://github.com/pytorch/executorch/blob/main/CONTRIBUTING.md#pull-requests) for ExecuTorch PR guidelines.

[PLEASE REMOVE] If this PR closes an issue, please add a `Fixes #<issue-id>` line.

[PLEASE REMOVE] If this PR introduces a fix or feature that should be the upcoming release notes, please add a "Release notes: <area>" label. For a list of available release notes labels, check out [CONTRIBUTING.md's Pull Requests](https://github.com/pytorch/executorch/blob/main/CONTRIBUTING.md#pull-requests).

### Test plan
[PLEASE REMOVE] How did you test this PR? Please write down any manual commands you used and note down tests that you have written if applicable.

### CODE REVIEW

**Comment by haowhsu-quic** [comment]
Hi @YIWENX14, I've created one draft PR (#16693) to address the SQNR issue of `static_llama`.
The result could be reproduced with:
```bash
python llm_numeric_match.py --checkpoint stories/stories110M.pt --params stories/params.json --tokenizer_model stories/tokenizer.model --prompt "Once upon a time" -s f3c0531 -b build-android/ -m SM8850
```
After finishing execution, there should be `llm_debug/llm_sqnr_debugging_graph.svg` for you to visualize the metrics of graph nodes.

For `8a8w` configuration, the SQNR drops below 20 in the first attention layer (wv linear):
image

If we change the quantizer to `16a4w`, we could have good SQNR for all layers and the final logits:
image

Current accuracy debugger is comparing the CPU goldens (QDQ parameters and floating-point operations) with dequantized HTP per-layer results. With this tool, we could quickly identify the problematic layers and apply higher precision solely with [quantization recipe](https://github.com/pytorch/executorch/blob/3b162950a516242da722c790cb466feb05cb299e/examples/qualcomm/oss_scripts/llama/static_llm_quant_recipe.py#L85).

Add @cccclai for awareness.

**Comment by YIWENX14** [comment]
Great work! Thanks for sharing the detailed per-layer SNR results—snr=49 looks good with 16a4w. This will be really helpful for debugging and narrowing down any accuracy issues.
